benchmarks/benchmark_template.cpp

```diff
@@ -780,6 +780,70 @@ static void BasicBench_ServoUrl(benchmark::State& state) {
 BENCHMARK(BasicBench_ServoUrl);
 #endif  // ADA_RUST
 
+static void BasicBench_AdaURL_URLPattern(benchmark::State& state) {
```

Comment:
Can you create a new benchmark file, preferable benchmarks/urlpattern.cpp, where we can benchmark the performance of ada::parse_url_pattern(), url_pattern::exec() and url_pattern::test(). Later, in a different PR we can test different types passed into the parse_url_pattern() method.